{"name": "Удалить товар", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame\n  (Session info: chrome=127.0.6533.88); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception\nStacktrace:\n#0 0x60d6c90c06ba <unknown>\n#1 0x60d6c8d90730 <unknown>\n#2 0x60d6c8d965c1 <unknown>\n#3 0x60d6c8d981f5 <unknown>\n#4 0x60d6c8d9827c <unknown>\n#5 0x60d6c8de1afa <unknown>\n#6 0x60d6c8de0d2a <unknown>\n#7 0x60d6c8dd4d66 <unknown>\n#8 0x60d6c8e02a52 <unknown>\n#9 0x60d6c8dd4719 <unknown>\n#10 0x60d6c8e02c1e <unknown>\n#11 0x60d6c8e211ca <unknown>\n#12 0x60d6c8e027f3 <unknown>\n#13 0x60d6c8dd2ec9 <unknown>\n#14 0x60d6c8dd391e <unknown>\n#15 0x60d6c90869eb <unknown>\n#16 0x60d6c908a972 <unknown>\n#17 0x60d6c9073e15 <unknown>\n#18 0x60d6c908b502 <unknown>\n#19 0x60d6c9058d2f <unknown>\n#20 0x60d6c90af578 <unknown>\n#21 0x60d6c90af750 <unknown>\n#22 0x60d6c90bf48c <unknown>\n#23 0x71450fbd2ac3 <unknown>", "trace": "browser = <selenium.webdriver.remote.webdriver.WebDriver (session=\"b8cb8a8957319cda203d08f25494c94c\")>, base_url = 'http://10.0.2.15:8081'\n\n    @allure.title(\"Удалить товар\")\n    def test_delete_product(browser, base_url):\n        admin_page = AdminPage(browser)\n        admin_page.open(base_url)\n        admin_page.login(\"user\", \"qwerty!123\")\n        admin_page.select_catalog()\n        admin_page.select_products()\n        HomePage(browser).scroll_down()\n        admin_page.to_the_last_page()\n>       admin_page.select_last_product()\n\ntests/test_start.py:120: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npage_objects/admin_page.py:110: in select_last_product\n    self.get_element(self.LAST_PRODUCT).click()\nvenv/lib/python3.12/site-packages/selenium/webdriver/remote/webelement.py:94: in click\n    self._execute(Command.CLICK_ELEMENT)\nvenv/lib/python3.12/site-packages/selenium/webdriver/remote/webelement.py:395: in _execute\n    return self._parent.execute(command, params)\nvenv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x788842746240>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"stale element reference\",\"message\":\"stale element reference: stale elemen...\\\\n#21 0x60d6c90af750 \\\\u003Cunknown>\\\\n#22 0x60d6c90bf48c \\\\u003Cunknown>\\\\n#23 0x71450fbd2ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame\nE         (Session info: chrome=127.0.6533.88); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception\nE       Stacktrace:\nE       #0 0x60d6c90c06ba <unknown>\nE       #1 0x60d6c8d90730 <unknown>\nE       #2 0x60d6c8d965c1 <unknown>\nE       #3 0x60d6c8d981f5 <unknown>\nE       #4 0x60d6c8d9827c <unknown>\nE       #5 0x60d6c8de1afa <unknown>\nE       #6 0x60d6c8de0d2a <unknown>\nE       #7 0x60d6c8dd4d66 <unknown>\nE       #8 0x60d6c8e02a52 <unknown>\nE       #9 0x60d6c8dd4719 <unknown>\nE       #10 0x60d6c8e02c1e <unknown>\nE       #11 0x60d6c8e211ca <unknown>\nE       #12 0x60d6c8e027f3 <unknown>\nE       #13 0x60d6c8dd2ec9 <unknown>\nE       #14 0x60d6c8dd391e <unknown>\nE       #15 0x60d6c90869eb <unknown>\nE       #16 0x60d6c908a972 <unknown>\nE       #17 0x60d6c9073e15 <unknown>\nE       #18 0x60d6c908b502 <unknown>\nE       #19 0x60d6c9058d2f <unknown>\nE       #20 0x60d6c90af578 <unknown>\nE       #21 0x60d6c90af750 <unknown>\nE       #22 0x60d6c90bf48c <unknown>\nE       #23 0x71450fbd2ac3 <unknown>\n\nvenv/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py:229: StaleElementReferenceException"}, "steps": [{"name": "Открываю http://10.0.2.15:8081/administration", "status": "passed", "start": 1725204873378, "stop": 1725204874705}, {"name": "Авторизую пользователя", "status": "passed", "parameters": [{"name": "username", "value": "'user'"}, {"name": "password", "value": "'qwerty!123'"}], "start": 1725204874714, "stop": 1725204876694}, {"name": "Перехожу в меню Каталог", "status": "passed", "start": 1725204876694, "stop": 1725204877746}, {"name": "Выбираю товар", "status": "passed", "start": 1725204877746, "stop": 1725204878592}, {"name": "Скроллю страницу вниз", "status": "passed", "start": 1725204878592, "stop": 1725204878671}, {"name": "Перехожу на последнюю страницу", "status": "passed", "start": 1725204878671, "stop": 1725204878903}, {"name": "Выбираю последний товар в списке", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame\n  (Session info: chrome=127.0.6533.88); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception\nStacktrace:\n#0 0x60d6c90c06ba <unknown>\n#1 0x60d6c8d90730 <unknown>\n#2 0x60d6c8d965c1 <unknown>\n#3 0x60d6c8d981f5 <unknown>\n#4 0x60d6c8d9827c <unknown>\n#5 0x60d6c8de1afa <unknown>\n#6 0x60d6c8de0d2a <unknown>\n#7 0x60d6c8dd4d66 <unknown>\n#8 0x60d6c8e02a52 <unknown>\n#9 0x60d6c8dd4719 <unknown>\n#10 0x60d6c8e02c1e <unknown>\n#11 0x60d6c8e211ca <unknown>\n#12 0x60d6c8e027f3 <unknown>\n#13 0x60d6c8dd2ec9 <unknown>\n#14 0x60d6c8dd391e <unknown>\n#15 0x60d6c90869eb <unknown>\n#16 0x60d6c908a972 <unknown>\n#17 0x60d6c9073e15 <unknown>\n#18 0x60d6c908b502 <unknown>\n#19 0x60d6c9058d2f <unknown>\n#20 0x60d6c90af578 <unknown>\n#21 0x60d6c90af750 <unknown>\n#22 0x60d6c90bf48c <unknown>\n#23 0x71450fbd2ac3 <unknown>\n\n", "trace": "  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/venv/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/page_objects/admin_page.py\", line 110, in select_last_product\n    self.get_element(self.LAST_PRODUCT).click()\n  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/venv/lib/python3.12/site-packages/selenium/webdriver/remote/webelement.py\", line 94, in click\n    self._execute(Command.CLICK_ELEMENT)\n  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/venv/lib/python3.12/site-packages/selenium/webdriver/remote/webelement.py\", line 395, in _execute\n    return self._parent.execute(command, params)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/venv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/venv/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "start": 1725204878903, "stop": 1725204879152}], "start": 1725204873378, "stop": 1725204879152, "uuid": "c70d8d95-c42c-4a84-a123-0bb17ac3dfcf", "historyId": "7d181faf2f27682f61cb2a0d2e84375e", "testCaseId": "7d181faf2f27682f61cb2a0d2e84375e", "fullName": "tests.test_start#test_delete_product", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_start"}, {"name": "host", "value": "fuenukkusu2501-VirtualBox"}, {"name": "thread", "value": "121555-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_start"}]}